app/controllers/provider_foreman_controller.rb

```diff
@@ -192,13 +179,7 @@ def cancel_provider_foreman
 
   def provider_foreman_form_fields
     assert_privileges("provider_foreman_edit_provider")
-    config_mgr = find_record(ManageIQ::Providers::ConfigurationManager, params[:id])
-    if config_mgr[:type] == "ManageIQ::Providers::AnsibleTower::ConfigurationManager"
-      provider = ManageIQ::Providers::AnsibleTower::Provider.find(config_mgr.provider_id)
-    else
-      provider = ManageIQ::Providers::Foreman::Provider.find(config_mgr.provider_id)
-    end
-    authentications = Authentication.where(:resource_id => provider[:id], :resource_type => "Provider")
+    authentications = ManageIQ::Providers::ConfigurationManager.find(params[:id]).provider.authentications
```

Comment:
seems we use `find_record` everywhere. Does it buy us anything? If not, I would love to eradicate it. :)